src/main/java/com/blackducksoftware/integration/hub/nexus/repository/task/PolicyCheckTask.java

```diff
@@ -59,43 +57,38 @@ protected String getRepositoryPathFieldId() {
     }
 
     @Override
-    protected Object doRun() throws Exception {
-        try {
-            final HubServiceHelper hubServiceHelper = new HubServiceHelper(new Slf4jIntLogger(logger), this.getParameters());
-            final List<Repository> repositoryList = getRepositoryRegistry().getRepositories();
-            final List<WalkerContext> contextList = new ArrayList<>();
+    protected String getAction() {
+        return "BLACKDUCK_HUB_POLICY_CHECK";
+    }
 
-            for (final Repository repository : repositoryList) {
-                contextList.add(createRepositoryWalker(repository, hubServiceHelper));
-            }
-            walkRepositories(contextList);
-        } catch (final Exception ex) {
-            logger.error("Error occurred during task execution {}", ex);
-        }
-        return null;
+    @Override
+    protected String getMessage() {
+        return "HUB-NEXUS-PLUGIN-POLICY-CHECK: Search for successfully scanned artifacts and check their policy";
     }
 
-    private WalkerContext createRepositoryWalker(final Repository repository, final HubServiceHelper hubServiceHelper) {
-        final ResourceStoreRequest request = new ResourceStoreRequest(getResourceStorePath(), true, false);
-        if (StringUtils.isBlank(request.getRequestPath())) {
-            request.setRequestPath(RepositoryItemUid.PATH_ROOT);
+    @Override
+    protected List<Repository> createRepositoryList() {
+        final List<Repository> repositoryList = new ArrayList<>();
+        final List<Repository> allRepositoryList = super.createRepositoryList();
+        for (final Repository repository : allRepositoryList) {
+            if (!repository.getRepositoryKind().isFacetAvailable(GroupRepository.class)) {
+                repositoryList.add(repository);
+            }
         }
 
-        request.setRequestLocalOnly(true);
-        final WalkerContext context = new DefaultWalkerContext(repository, request);
-        getLogger().info("Creating walker for repository {}", repository.getName());
-        context.getProcessors().add(new PolicyRepositoryWalker(getEventBus(), new ItemAttributesHelper(attributesHandler), getParameters(), hubServiceHelper));
-        return context;
+        return repositoryList;
     }
 
     @Override
-    protected String getAction() {
-        return "BLACKDUCK_HUB_POLICY_CHECK";
+    protected AbstractWalkerProcessor getRepositoryWalker() {
+        final ExecutorService executorService = parallelEventProcessor.createExecutorService();
+        parallelEventProcessor.setExecutorService(executorService);
```

Comment:
The setExecutorService method you may want to make protected so it can be called by classes in the same package.  Namely the test classes but it isn't accessible publicly.  Then in the createExecutorService() method call the setExecutorService method.  I will add corresponding comments in parallelEventProcessor.